bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitTaggedItem.java

```diff
@@ -415,14 +417,34 @@ public double getConfigurationAsDouble(String key, double defaultValue) {
      * @param defaultValue default value
      * @return value
      */
-    public QuantityType getConfigurationAsQuantity(String key, QuantityType defaultValue) {
+    public QuantityType<?> getConfigurationAsQuantity(String key, QuantityType defaultValue,
+            boolean relativeConversion) {
         String stringValue = getConfiguration(key, new String());
         if (stringValue.isEmpty()) {
             return defaultValue;
         }
         var parsedValue = new QuantityType(stringValue);
-        var convertedValue = parsedValue.toInvertibleUnit(defaultValue.getUnit());
-        // not convertible? just assume it's in the expected unit
+        QuantityType<?> convertedValue;
+
+        if (relativeConversion) {
+            convertedValue = parsedValue.toUnitRelative(defaultValue.getUnit());
+        } else {
+            convertedValue = parsedValue.toInvertibleUnit(defaultValue.getUnit());
+        }
+        // not convertible? just assume it's in the item's unit
+        if (convertedValue == null) {
+            Unit unit;
+            if (getBaseItem() instanceof NumberItem && (unit = ((NumberItem) getBaseItem()).getUnit()) != null) {
```

Comment:
i have not seen the assignment (=) in IF statement that often in Java. but remember it from old good C/C++ days.
but if it works, why not. probably saves another IF.